### PR TITLE
Seperate docs into docs and docs-gallery

### PR DESCRIPTION
**Pull request by nabobalis**
Following the problems of installing mplcario, I have separate docs into docs and docs-gallery. Only docs is installed for the dev extra.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -126,6 +121,12 @@ docs =
   sphinxext-opengraph
   sunpy-sphinx-theme>=2.0.0rc1
   sphinx-hoverxref
+docs-gallery =
```
Comment (nabobalis) [inline setup.cfg:124]: I think thius is everything.

**Review by alasdairwilson — APPROVED** [review]

**Review by dstansby — CHANGES_REQUESTED** [review]
👍  - the fact that you now need to install the [docs-gallery] extra to build the gallery should be documented somewhere in the developer guide.

File: docs/dev_guide/contents/newcomers.rst [hunk 2]
```diff
@@ -143,6 +143,8 @@ Next we will want to setup the conda environment and we will need to add the `co
 .. code:: bash
 
     $ conda config --add channels conda-forge
+    # Note you might need to add python=<version> if a new one has come out very recently.
```
Comment (Cadair) [inline docs/dev_guide/contents/newcomers.rst:146]:
```suggestion
    # Note you might need to add python=<version> if a new release of Python has come out very recently.
```

**Comment by nabobalis** [comment]
> 👍 - the fact that you now need to install the [docs-gallery] extra to build the gallery should be documented somewhere in the developer guide.

Should be fixed hopefully.

**Review by Cadair — APPROVED** [review]

**Review by dstansby — APPROVED** [review]

**Pull request merged**